language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/VariableDeclarationNodeContext.java

```diff
@@ -45,21 +50,42 @@ public VariableDeclarationNodeContext() {
     @Override
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, VariableDeclarationNode node)
             throws LSCompletionException {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
         if (node.initializer().isPresent() && onExpressionContext(context, node)) {
-            List<LSCompletionItem> completionItems
-                    = this.initializerContextCompletions(context, node.typedBindingPattern().typeDescriptor());
+            ExpressionNode initializer = node.initializer().get();
+            if (onSuggestionsAfterQualifiers(context, initializer)) {
+                /*
+                    Covers the following
+                    x = <qualifier(s)> <cursor>
+                    x = <qualifier(s)>  x<cursor>
+                    currently the qualifier can be isolated/transactional.
+                 */
+                List<Token> qualifiers = initializer.leadingInvalidTokens();
+                Token lastQualifier = qualifiers.get(qualifiers.size() - 1);
+                Set<SyntaxKind> qualKinds = qualifiers.stream().map(Node::kind).collect(Collectors.toSet());
+                completionItems.addAll(getCompletionItemsOnQualifiers(qualKinds, lastQualifier, context));
```

Comment:
We are duplicating this code segment. Let's clean this if else ladder

Reply:
Also removing the routing to the parent node as we have handled the isolation case now. This could take some time, as there are many test cases failing after removing that

Reply:
>We are duplicating this code segment. Let's clean this if else ladder
Done